### PR TITLE
Webdev 923 reorganise paragraph component

### PR DESCRIPTION
**Pull request by Christine-horrocks**

### CODE REVIEW

File: app/serializers/component_serializer/paragraph_component_serializer.rb [hunk 1]
```diff
@@ -23,15 +21,7 @@ def name
...
-      @content.map do |content|
-        {}.tap do |hash|
-          hash[:content] = content[:content]
-
-          data = content.try(:except, :content)
-
-          hash[:data] = data if data.present?
-        end
-      end
+      @content if @content
```
Comment (mattrayner) [inline app/serializers/component_serializer/paragraph_component_serializer.rb]: What if you provide no content, will the serialiser still work with blank content?

Comment (Christine-horrocks) [reply app/serializers/component_serializer/paragraph_component_serializer.rb]: You are right, It should not work if it is not given content. That would be the minimum requirement. I will take away that conditional.

File: app/controllers/concerns/content_data_helper.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module ContentDataHelper
+  # Helper method to create hash for any part of a component or partial which needs a content and data hash to send through a translation key with multiple data items.
+  # The content value is the translation key.
```
Comment (katylouise) [inline app/controllers/concerns/content_data_helper.rb]: Could you use the yard style of documenting here?

File: app/controllers/concerns/content_data_helper.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  # The data hash contains key/value pairs where the key is also the key inside curly braces in the string to be interpolated. The values are the data from the database which will be inserted into the string.
+  # Should this be called the data interpolation helper??
```
Comment (katylouise) [inline app/controllers/concerns/content_data_helper.rb]: Have you answered this question for yourself?

File: app/controllers/concerns/content_data_helper.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+module ContentDataHelper
+  # Helper method to create hash for any part of a component or partial which needs a content and data hash to send through a translation key with multiple data items.
+  # The data hash contains key/value pairs where the key is also the key inside curly braces in the string to be interpolated. The values are the data from the database which will be inserted into the string.
+  # @param [String] content is the translation key to be added as the content value
+  # @param [Symbol] key is the key
```
Comment (katylouise) [inline app/controllers/concerns/content_data_helper.rb]: There are only two params to document: content and **data.  You could do something like:
`@param [String] content ...`
`@params [Array<Hash>] data ...`

**Pull request merged**